src/rust/rust-proto/src/graplinc/grapl/api/graph_query_service/v1beta1/client.rs

```diff
@@ -20,40 +31,77 @@ pub enum GraphQueryClientError {
     SerDeError(#[from] SerDeError),
     #[error("Status {0}")]
     Status(#[from] Status),
-    #[error("ConnectError")]
-    ConnectError(tonic::transport::Error),
+    #[error("Rejected")]
+    Rejected,
+    #[error("Elapsed")]
+    Elapsed,
+}
+
+impl From<client_executor::Error<GraphQueryClientError>> for GraphQueryClientError {
+    fn from(e: client_executor::Error<GraphQueryClientError>) -> Self {
+        match e {
+            client_executor::Error::Rejected => GraphQueryClientError::Rejected,
+            client_executor::Error::Elapsed => GraphQueryClientError::Elapsed,
+            client_executor::Error::Inner(e) => e,
+        }
+    }
 }
 
 #[derive(Clone)]
 pub struct GraphQueryClient {
     inner: GraphQueryServiceClient<tonic::transport::Channel>,
+    executor: Executor,
 }
 
 impl GraphQueryClient {
-    pub async fn connect<T>(endpoint: T) -> Result<Self, GraphQueryClientError>
+    pub async fn connect<T>(endpoint: T) -> Result<Self, ConnectError>
     where
-        T: TryInto<tonic::transport::Endpoint>,
-        T::Error: std::error::Error + Send + Sync + 'static,
+        T: Clone,
+        tonic::transport::Endpoint: TryFrom<T> + Clone,
+        <T as TryInto<tonic::transport::Endpoint>>::Error:
+            std::error::Error + Send + Sync + 'static,
     {
+        let executor = Executor::new(ExecutorConfig::new(Duration::from_secs(30)));
+        let proto_client = create_proto_client!(
+            executor,
+            GraphQueryServiceClient<tonic::transport::Channel>,
+            endpoint,
+        );
+
         Ok(GraphQueryClient {
-            inner: GraphQueryServiceClient::connect(endpoint)
-                .await
-                .map_err(GraphQueryClientError::ConnectError)?,
+            inner: proto_client,
```

Comment:
Rename `inner` to `proto_client` - i suspect that's why the macro would have been failing.